mlir/include/mlir/Transforms/DialectConversion.h

```diff
@@ -1070,6 +1072,30 @@ class PDLConversionConfig final {
 
 #endif // MLIR_ENABLE_PDL_IN_PATTERNMATCH
 
+//===----------------------------------------------------------------------===//
+// ConversionConfig
+//===----------------------------------------------------------------------===//
+
+/// Dialect conversion configuration.
+struct ConversionConfig {
+  /// An optional callback used to notify about match failure diagnostics during
+  /// the conversion. Diagnostics are only reported to this callback may only be
```

Comment:
"Diagnostics are only reported to this callback may only be available in debug mode" - this reads a bit weird.